assets/js/components/channels/ChannelIndex.jsx

```diff
@@ -37,6 +57,31 @@ class ChannelIndex extends Component<any> {
   toggleProvider(provider, index, checked) {
     if (checked) {
       $(`#${provider}Modal-${index}`).modal("open")
+      this.setState({
+        modalLoading: true,
+        modalSurveys: [],
+        modalProvider: provider,
+        modalIndex: index,
+      })
+      const { baseUrl } = config[provider][index]
+      api.fetchActiveSurveys(provider, baseUrl)
+      .then((response) => {
+        const surveys = response || []
+        this.setState({
+          modalLoading: false,
+          modalSurveys: surveys,
+          modalProvider: provider,
+          modalIndex: index,
+        })
+      })
+      .catch(() => {
+        this.setState({
+          modalLoading: false,
+          modalSurveys: [],
+          modalProvider: provider,
+          modalIndex: index,
+        })
```

Comment:
This is silently dismissing an exception. Shouldn't we let it throw (so the user gets an automatic error message popup)? Or set `modalSurveys` to `null` ("error, beware") and tell it apart from an empty list ("no active channels, safe to delete provider").

Reply:
> I'm showing a generic error message an logging the error to the browser's console

Isn't that Surveda's default behaviour if we simply don't catch the exception? There's some standard component there that shows exceptions as a notification bubble, not sure if that applies here too.

Reply:
I've removed the catch handler and changes related.